--- centaur.act.old	2012-11-29 13:26:32.062325973 -0500
+++ centaur.act	2012-11-29 13:26:55.228400406 -0500
@@ -187,7 +187,7 @@ CAUSE_EFFECT {
   WATCH=[REG(0x0301060C)]  #MBMSR=0301060C
   CAUSE: TARGET=[REG(0x0301060C)] OP=[BIT,OFF] BIT=[0]   # Maint Cmd Complete
   CAUSE: TARGET=[REG(0x0301060F)] OP=[BIT,ON] BIT=[11]   # Enable Special attn
-  #EFFECT: TARGET=[REG(0x03010611)] OP=[BIT,ON] BIT=[0]   # Set Special attn bit
+  EFFECT: TARGET=[REG(0x03010611)] OP=[BIT,ON] BIT=[0]   # Set Special attn bit
 }

 #Set command in progress
@@ -223,7 +223,7 @@ CAUSE_EFFECT {
   WATCH=[REG(0x03010E0C)]  #MBMSR=03010E0C
   CAUSE: TARGET=[REG(0x03010E0C)] OP=[BIT,OFF] BIT=[0]   # Maint Cmd Complete
   CAUSE: TARGET=[REG(0x03010E0F)] OP=[BIT,ON] BIT=[11]   # Enable Special attn
-  #EFFECT: TARGET=[REG(0x03010E11)] OP=[BIT,ON] BIT=[0]   # Set Special attn bit
+  EFFECT: TARGET=[REG(0x03010E11)] OP=[BIT,ON] BIT=[0]   # Set Special attn bit
 }

 #***************************************
