--- p8.chip.build	2013-08-20 12:00:22.521871799 -0500
+++ p8.chip.new	2013-08-20 11:04:02.480135466 -0500
@@ -37,6 +37,7 @@
 #        SW201429  pacharya  05/06/13  remove s1_ex.act
 #        D884154   dthorn    05/22/13  temporarily add missing rings
 #        F884180   dkodihal  06/06/13  Updated SCANRANGE with fastarray rings
+#        D892792   jknight   08/20/13  Updated SCOM ranges
 
 VERSION 1               # .chip file format ID
 TYPE=p8                 # Chip Type
@@ -156,9 +157,10 @@
 SCOMRANGE    # Value, Mask, Clocks, Logical Address
 
     # Scom Ranges for ex14 
-    0x1E040000, 0xFFF40000, 0x80000000 00000003, 0xFFC5001E  # ex14prv, chiplet vital, chip vital 
-    0x1E030000, 0xFFF30000, 0x00000000 00000003, 0xFFC5001E  # ex14vital, chiplet vital, chip vital 
-    0x1E020000, 0xFFF20000, 0x80000000 00000003, 0xFFC5001E  # ex14oha, chiplet vital, chip vital 
+    0x1E0F0000, 0xFFFF0000, 0x00000000 00000001, 0xFFC5001E  # chip vital 
+    0x1E040000, 0xFFFF0000, 0x80000000 00000003, 0xFFC5001E  # ex14prv, chiplet vital, chip vital 
+    0x1E030000, 0xFFFF0000, 0x00000000 00000003, 0xFFC5001E  # ex14vital, chiplet vital, chip vital 
+    0x1E020000, 0xFFFF0000, 0x80000000 00000003, 0xFFC5001E  # ex14oha, chiplet vital, chip vital 
     0x1E013C00, 0xFFFFFC00, 0x60000000 00000003, 0xFFC5001E  # ex14pcsec, chiplet vital, chip vital 
     0x1E013000, 0xFFFF3000, 0x60000000 00000003, 0xFFC5001E  # ex14pc, chiplet vital, chip vital 
     0x1E012C00, 0xFFFFFC00, 0xE0000000 00000003, 0xFFC5001E  # ex14trapc, chiplet vital, chip vital 
@@ -172,9 +174,10 @@
     0x1E000000, 0xFE000000, 0x00000000 00000003, 0xFFC5001E  # ex14vital, chiplet vital, chip vital 
 
     # Scom Ranges for ex13 
-    0x1D040000, 0xFFF40000, 0x80000000 00000003, 0xFFC5001D  # ex13prv, chiplet vital, chip vital 
-    0x1D030000, 0xFFF30000, 0x00000000 00000003, 0xFFC5001D  # ex13vital, chiplet vital, chip vital 
-    0x1D020000, 0xFFF20000, 0x80000000 00000003, 0xFFC5001D  # ex13oha, chiplet vital, chip vital 
+    0x1D0F0000, 0xFFFF0000, 0x00000000 00000001, 0xFFC5001D  # chip vital 
+    0x1D040000, 0xFFFF0000, 0x80000000 00000003, 0xFFC5001D  # ex13prv, chiplet vital, chip vital 
+    0x1D030000, 0xFFFF0000, 0x00000000 00000003, 0xFFC5001D  # ex13vital, chiplet vital, chip vital 
+    0x1D020000, 0xFFFF0000, 0x80000000 00000003, 0xFFC5001D  # ex13oha, chiplet vital, chip vital 
     0x1D013C00, 0xFFFFFC00, 0x60000000 00000003, 0xFFC5001D  # ex13pcsec, chiplet vital, chip vital 
     0x1D013000, 0xFFFF3000, 0x60000000 00000003, 0xFFC5001D  # ex13pc, chiplet vital, chip vital 
     0x1D012C00, 0xFFFFFC00, 0xE0000000 00000003, 0xFFC5001D  # ex13trapc, chiplet vital, chip vital 
@@ -187,10 +190,11 @@
     0x1D010400, 0xFFFFF400, 0xE0000000 00000003, 0xFFC5001D  # ex13traeco, chiplet vital, chip vital 
     0x1D000000, 0xFD000000, 0x00000000 00000003, 0xFFC5001D  # ex13vital, chiplet vital, chip vital 
 
-    # Scom Ranges for ex12 
-    0x1C040000, 0xFFF40000, 0x80000000 00000003, 0xFFC5001C  # ex12prv, chiplet vital, chip vital 
-    0x1C030000, 0xFFF30000, 0x00000000 00000003, 0xFFC5001C  # ex12vital, chiplet vital, chip vital 
-    0x1C020000, 0xFFF20000, 0x80000000 00000003, 0xFFC5001C  # ex12oha, chiplet vital, chip vital 
+    # Scom Ranges for ex12
+    0x1C0F0000, 0xFFFF0000, 0x00000000 00000001, 0xFFC5001C  # chip vital 
+    0x1C040000, 0xFFFF0000, 0x80000000 00000003, 0xFFC5001C  # ex12prv, chiplet vital, chip vital 
+    0x1C030000, 0xFFFF0000, 0x00000000 00000003, 0xFFC5001C  # ex12vital, chiplet vital, chip vital 
+    0x1C020000, 0xFFFF0000, 0x80000000 00000003, 0xFFC5001C  # ex12oha, chiplet vital, chip vital 
     0x1C013C00, 0xFFFFFC00, 0x60000000 00000003, 0xFFC5001C  # ex12pcsec, chiplet vital, chip vital 
     0x1C013000, 0xFFFF3000, 0x60000000 00000003, 0xFFC5001C  # ex12pc, chiplet vital, chip vital 
     0x1C012C00, 0xFFFFFC00, 0xE0000000 00000003, 0xFFC5001C  # ex12trapc, chiplet vital, chip vital 
@@ -203,10 +207,11 @@
     0x1C010400, 0xFFFFF400, 0xE0000000 00000003, 0xFFC5001C  # ex12traeco, chiplet vital, chip vital 
     0x1C000000, 0xFC000000, 0x00000000 00000003, 0xFFC5001C  # ex12vital, chiplet vital, chip vital 
 
-    # Scom Ranges for ex11 
-    0x1B040000, 0xFFF40000, 0x80000000 00000003, 0xFFC5001B  # ex11prv, chiplet vital, chip vital 
-    0x1B030000, 0xFFF30000, 0x00000000 00000003, 0xFFC5001B  # ex11vital, chiplet vital, chip vital 
-    0x1B020000, 0xFFF20000, 0x80000000 00000003, 0xFFC5001B  # ex11oha, chiplet vital, chip vital 
+    # Scom Ranges for ex11
+    0x1B0F0000, 0xFFFF0000, 0x00000000 00000001, 0xFFC5001B  # chip vital 
+    0x1B040000, 0xFFFF0000, 0x80000000 00000003, 0xFFC5001B  # ex11prv, chiplet vital, chip vital 
+    0x1B030000, 0xFFFF0000, 0x00000000 00000003, 0xFFC5001B  # ex11vital, chiplet vital, chip vital 
+    0x1B020000, 0xFFFF0000, 0x80000000 00000003, 0xFFC5001B  # ex11oha, chiplet vital, chip vital 
     0x1B013C00, 0xFFFFFC00, 0x60000000 00000003, 0xFFC5001B  # ex11pcsec, chiplet vital, chip vital 
     0x1B013000, 0xFFFF3000, 0x60000000 00000003, 0xFFC5001B  # ex11pc, chiplet vital, chip vital 
     0x1B012C00, 0xFFFFFC00, 0xE0000000 00000003, 0xFFC5001B  # ex11trapc, chiplet vital, chip vital 
@@ -219,9 +224,10 @@
     0x1B010400, 0xFFFFF400, 0xE0000000 00000003, 0xFFC5001B  # ex11traeco, chiplet vital, chip vital 
 
     # Scom Ranges for ex10 
-    0x1A040000, 0xFFF40000, 0x80000000 00000003, 0xFFC5001A  # ex10prv, chiplet vital, chip vital 
-    0x1A030000, 0xFFF30000, 0x00000000 00000003, 0xFFC5001A  # ex10vital, chiplet vital, chip vital 
-    0x1A020000, 0xFFF20000, 0x80000000 00000003, 0xFFC5001A  # ex10oha, chiplet vital, chip vital 
+    0x1A0F0000, 0xFFFF0000, 0x00000000 00000001, 0xFFC5001A  # chip vital 
+    0x1A040000, 0xFFFF0000, 0x80000000 00000003, 0xFFC5001A  # ex10prv, chiplet vital, chip vital 
+    0x1A030000, 0xFFFF0000, 0x00000000 00000003, 0xFFC5001A  # ex10vital, chiplet vital, chip vital 
+    0x1A020000, 0xFFFF0000, 0x80000000 00000003, 0xFFC5001A  # ex10oha, chiplet vital, chip vital 
     0x1A013C00, 0xFFFFFC00, 0x60000000 00000003, 0xFFC5001A  # ex10pcsec, chiplet vital, chip vital 
     0x1A013000, 0xFFFF3000, 0x60000000 00000003, 0xFFC5001A  # ex10pc, chiplet vital, chip vital 
     0x1A012C00, 0xFFFFFC00, 0xE0000000 00000003, 0xFFC5001A  # ex10trapc, chiplet vital, chip vital 
@@ -234,9 +240,10 @@
     0x1A010400, 0xFFFFF400, 0xE0000000 00000003, 0xFFC5001A  # ex10traeco, chiplet vital, chip vital 
 
     # Scom Ranges for ex09 
-    0x19040000, 0xFFF40000, 0x80000000 00000003, 0xFFC50019  # ex09prv, chiplet vital, chip vital 
-    0x19030000, 0xFFF30000, 0x00000000 00000003, 0xFFC50019  # ex09vital, chiplet vital, chip vital 
-    0x19020000, 0xFFF20000, 0x80000000 00000003, 0xFFC50019  # ex09oha, chiplet vital, chip vital 
+    0x190F0000, 0xFFFF0000, 0x00000000 00000001, 0xFFC50019  # chip vital 
+    0x19040000, 0xFFFF0000, 0x80000000 00000003, 0xFFC50019  # ex09prv, chiplet vital, chip vital 
+    0x19030000, 0xFFFF0000, 0x00000000 00000003, 0xFFC50019  # ex09vital, chiplet vital, chip vital 
+    0x19020000, 0xFFFF0000, 0x80000000 00000003, 0xFFC50019  # ex09oha, chiplet vital, chip vital 
     0x19013C00, 0xFFFFFC00, 0x60000000 00000003, 0xFFC50019  # ex09pcsec, chiplet vital, chip vital 
     0x19013000, 0xFFFF3000, 0x60000000 00000003, 0xFFC50019  # ex09pc, chiplet vital, chip vital 
     0x19012C00, 0xFFFFFC00, 0xE0000000 00000003, 0xFFC50019  # ex09trapc, chiplet vital, chip vital 
@@ -249,9 +256,10 @@
     0x19010400, 0xFFFFF400, 0xE0000000 00000003, 0xFFC50019  # ex09traeco, chiplet vital, chip vital 
 
     # Scom Ranges for ex06 
-    0x16040000, 0xFFF40000, 0x80000000 00000003, 0xFFC50016  # ex06prv, chiplet vital, chip vital 
-    0x16030000, 0xFFF30000, 0x00000000 00000003, 0xFFC50016  # ex06vital, chiplet vital, chip vital 
-    0x16020000, 0xFFF20000, 0x80000000 00000003, 0xFFC50016  # ex06oha, chiplet vital, chip vital 
+    0x160F0000, 0xFFFF0000, 0x00000000 00000001, 0xFFC50016  # chip vital 
+    0x16040000, 0xFFFF0000, 0x80000000 00000003, 0xFFC50016  # ex06prv, chiplet vital, chip vital 
+    0x16030000, 0xFFFF0000, 0x00000000 00000003, 0xFFC50016  # ex06vital, chiplet vital, chip vital 
+    0x16020000, 0xFFFF0000, 0x80000000 00000003, 0xFFC50016  # ex06oha, chiplet vital, chip vital 
     0x16013C00, 0xFFFFFC00, 0x60000000 00000003, 0xFFC50016  # ex06pcsec, chiplet vital, chip vital 
     0x16013000, 0xFFFF3000, 0x60000000 00000003, 0xFFC50016  # ex06pc, chiplet vital, chip vital 
     0x16012C00, 0xFFFFFC00, 0xE0000000 00000003, 0xFFC50016  # ex06trapc, chiplet vital, chip vital 
@@ -265,9 +273,10 @@
     0x16000000, 0xF6000000, 0x00000000 00000003, 0xFFC50016  # ex06vital, chiplet vital, chip vital 
 
     # Scom Ranges for ex05 
-    0x15040000, 0xFFF40000, 0x80000000 00000003, 0xFFC50015  # ex05prv, chiplet vital, chip vital 
-    0x15030000, 0xFFF30000, 0x00000000 00000003, 0xFFC50015  # ex05vital, chiplet vital, chip vital 
-    0x15020000, 0xFFF20000, 0x80000000 00000003, 0xFFC50015  # ex05oha, chiplet vital, chip vital 
+    0x150F0000, 0xFFFF0000, 0x00000000 00000001, 0xFFC50015  # chip vital 
+    0x15040000, 0xFFFF0000, 0x80000000 00000003, 0xFFC50015  # ex05prv, chiplet vital, chip vital 
+    0x15030000, 0xFFFF0000, 0x00000000 00000003, 0xFFC50015  # ex05vital, chiplet vital, chip vital 
+    0x15020000, 0xFFFF0000, 0x80000000 00000003, 0xFFC50015  # ex05oha, chiplet vital, chip vital 
     0x15013C00, 0xFFFFFC00, 0x60000000 00000003, 0xFFC50015  # ex05pcsec, chiplet vital, chip vital 
     0x15013000, 0xFFFF3000, 0x60000000 00000003, 0xFFC50015  # ex05pc, chiplet vital, chip vital 
     0x15012C00, 0xFFFFFC00, 0xE0000000 00000003, 0xFFC50015  # ex05trapc, chiplet vital, chip vital 
@@ -280,10 +289,11 @@
     0x15010400, 0xFFFFF400, 0xE0000000 00000003, 0xFFC50015  # ex05traeco, chiplet vital, chip vital 
     0x15000000, 0xF5000000, 0x00000000 00000003, 0xFFC50015  # ex05vital, chiplet vital, chip vital 
 
-    # Scom Ranges for ex04 
-    0x14040000, 0xFFF40000, 0x80000000 00000003, 0xFFC50014  # ex04prv, chiplet vital, chip vital 
-    0x14030000, 0xFFF30000, 0x00000000 00000003, 0xFFC50014  # ex04vital, chiplet vital, chip vital 
-    0x14020000, 0xFFF20000, 0x80000000 00000003, 0xFFC50014  # ex04oha, chiplet vital, chip vital 
+    # Scom Ranges for ex04
+    0x140F0000, 0xFFFF0000, 0x00000000 00000001, 0xFFC50014  # chip vital 
+    0x14040000, 0xFFFF0000, 0x80000000 00000003, 0xFFC50014  # ex04prv, chiplet vital, chip vital 
+    0x14030000, 0xFFFF0000, 0x00000000 00000003, 0xFFC50014  # ex04vital, chiplet vital, chip vital 
+    0x14020000, 0xFFFF0000, 0x80000000 00000003, 0xFFC50014  # ex04oha, chiplet vital, chip vital 
     0x14013C00, 0xFFFFFC00, 0x60000000 00000003, 0xFFC50014  # ex04pcsec, chiplet vital, chip vital 
     0x14013000, 0xFFFF3000, 0x60000000 00000003, 0xFFC50014  # ex04pc, chiplet vital, chip vital 
     0x14012C00, 0xFFFFFC00, 0xE0000000 00000003, 0xFFC50014  # ex04trapc, chiplet vital, chip vital 
@@ -297,9 +307,10 @@
     0x14000000, 0xF4000000, 0x00000000 00000003, 0xFFC50014  # ex04vital, chiplet vital, chip vital 
 
     # Scom Ranges for ex03 
-    0x13040000, 0xFFF40000, 0x80000000 00000003, 0xFFC50013  # ex03prv, chiplet vital, chip vital 
-    0x13030000, 0xFFF30000, 0x00000000 00000003, 0xFFC50013  # ex03vital, chiplet vital, chip vital 
-    0x13020000, 0xFFF20000, 0x80000000 00000003, 0xFFC50013  # ex03oha, chiplet vital, chip vital 
+    0x130F0000, 0xFFFF0000, 0x00000000 00000001, 0xFFC50013  # chip vital 
+    0x13040000, 0xFFFF0000, 0x80000000 00000003, 0xFFC50013  # ex03prv, chiplet vital, chip vital 
+    0x13030000, 0xFFFF0000, 0x00000000 00000003, 0xFFC50013  # ex03vital, chiplet vital, chip vital 
+    0x13020000, 0xFFFF0000, 0x80000000 00000003, 0xFFC50013  # ex03oha, chiplet vital, chip vital 
     0x13013C00, 0xFFFFFC00, 0x60000000 00000003, 0xFFC50013  # ex03pcsec, chiplet vital, chip vital 
     0x13013000, 0xFFFF3000, 0x60000000 00000003, 0xFFC50013  # ex03pc, chiplet vital, chip vital 
     0x13012C00, 0xFFFFFC00, 0xE0000000 00000003, 0xFFC50013  # ex03trapc, chiplet vital, chip vital 
@@ -312,9 +323,10 @@
     0x13010400, 0xFFFFF400, 0xE0000000 00000003, 0xFFC50013  # ex03traeco, chiplet vital, chip vital 
 
     # Scom Ranges for ex02 
-    0x12040000, 0xFFF40000, 0x80000000 00000003, 0xFFC50012  # ex02prv, chiplet vital, chip vital 
-    0x12030000, 0xFFF30000, 0x00000000 00000003, 0xFFC50012  # ex02vital, chiplet vital, chip vital 
-    0x12020000, 0xFFF20000, 0x80000000 00000003, 0xFFC50012  # ex02oha, chiplet vital, chip vital 
+    0x120F0000, 0xFFFF0000, 0x00000000 00000001, 0xFFC50012  # chip vital 
+    0x12040000, 0xFFFF0000, 0x80000000 00000003, 0xFFC50012  # ex02prv, chiplet vital, chip vital 
+    0x12030000, 0xFFFF0000, 0x00000000 00000003, 0xFFC50012  # ex02vital, chiplet vital, chip vital 
+    0x12020000, 0xFFFF0000, 0x80000000 00000003, 0xFFC50012  # ex02oha, chiplet vital, chip vital 
     0x12013C00, 0xFFFFFC00, 0x60000000 00000003, 0xFFC50012  # ex02pcsec, chiplet vital, chip vital 
     0x12013000, 0xFFFF3000, 0x60000000 00000003, 0xFFC50012  # ex02pc, chiplet vital, chip vital 
     0x12012C00, 0xFFFFFC00, 0xE0000000 00000003, 0xFFC50012  # ex02trapc, chiplet vital, chip vital 
@@ -327,9 +339,10 @@
     0x12010400, 0xFFFFF400, 0xE0000000 00000003, 0xFFC50012  # ex02traeco, chiplet vital, chip vital 
 
     # Scom Ranges for ex01 
-    0x11040000, 0xFFF40000, 0x80000000 00000003, 0xFFC50011  # ex01prv, chiplet vital, chip vital 
-    0x11030000, 0xFFF30000, 0x00000000 00000003, 0xFFC50011  # ex01vital, chiplet vital, chip vital 
-    0x11020000, 0xFFF20000, 0x80000000 00000003, 0xFFC50011  # ex01oha, chiplet vital, chip vital 
+    0x110F0000, 0xFFFF0000, 0x00000000 00000001, 0xFFC50011  # chip vital 
+    0x11040000, 0xFFFF0000, 0x80000000 00000003, 0xFFC50011  # ex01prv, chiplet vital, chip vital 
+    0x11030000, 0xFFFF0000, 0x00000000 00000003, 0xFFC50011  # ex01vital, chiplet vital, chip vital 
+    0x11020000, 0xFFFF0000, 0x80000000 00000003, 0xFFC50011  # ex01oha, chiplet vital, chip vital 
     0x11013C00, 0xFFFFFC00, 0x60000000 00000003, 0xFFC50011  # ex01pcsec, chiplet vital, chip vital 
     0x11013000, 0xFFFF3000, 0x60000000 00000003, 0xFFC50011  # ex01pc, chiplet vital, chip vital 
     0x11012C00, 0xFFFFFC00, 0xE0000000 00000003, 0xFFC50011  # ex01trapc, chiplet vital, chip vital 
