--- .bld.passes.mk	2015-03-23 14:38:42.000000000 -0500
+++ bld.passes.mk	2015-03-23 14:47:33.000000000 -0500
@@ -135,7 +135,9 @@
                  ${BUILDJAVADOCS:D${JAVADOC_AFTER_JAVAC:DJAVADOC}} \
                  JAR EARWAR
 .else
+# PJE - add setup pass to build
 _BUILD_PASSES_ = ${BUILDJAVADOCS:D${JAVADOC_AFTER_JAVAC:UJAVADOC}} \
+                 SETUP \
                  EXPINC OBJECTS \
                  ${BUILDJAVADOCS:D${JAVADOC_AFTER_JAVAC:DJAVADOC}} \
                  EXPLIB ${SHLIB_DURING_EXPSHLIB:DEXPSHLIB} STANDARD
