--- s1.act.old  2013-01-02 13:32:25.000000000 -0600
+++ s1.act      2013-01-03 07:02:03.000000000 -0600
@@ -31,6 +31,7 @@
 #  mc01  F864674   missyc    12/17/12  Added PLL Lock actions
 #        F864543   thi       12/19/12  Fix action file error
 #        D864795   dsanner   12/20/12  Fix clock actions
+#        F865077   dsanner   01/02/13  Updated exitCacheContainedMode

 CAUSE_EFFECT {
     LABEL=[Assert all logic clock domains off when chip logic power asserted off]
@@ -483,9 +484,10 @@
 # Note: Memory regions must have been created prior to this
 # This will exit cache contained when non mirror bar is writen to any EX core.
 # This only happens on the master core
-CAUSE_EFFECT CHIPLETS ex {
+CAUSE_EFFECT {
   LABEL=[Exit Cache Contained mode]
-  WATCH=[REG(MYCHIPLET,0x0001080b)]
+  WATCH=[REG(0x0201340C)]
+  CAUSE: TARGET=[LOGIC(0xFF0CC005)] OP=[BIT,ON] BIT=[0] # Master chip
   EFFECT: TARGET=[MODULE(exitCacheContainedMode)] OP=[MODULECALL]
 }
