--- simu.b0907a/data/simicsInfo	2016-09-19 12:58:13.377973614 -0500
+++ simu/data/simicsInfo	2016-09-19 12:56:08.167930877 -0500
@@ -76,5 +76,5 @@
 #####################################
 WSALIAS DEFAULT BBLEVEL none
 WSALIAS DEFAULT SIMICSLEVEL env/vtechf/simics-4.8.0/simics-4.8.140/bin
-WSALIAS DEFAULT FIPSLEVEL env/simf/simics-4.8.0/simics-4.8.140/fips/fld36/fi160905a900.48
-WSALIAS DEFAULT MAMBOLEVEL env/mamboa/2016_07_27__4.8
+WSALIAS DEFAULT FIPSLEVEL env/simf/simics-4.8.0/simics-4.8.140/fips/fld36/fi160914a900.48
+WSALIAS DEFAULT MAMBOLEVEL env/mamboa/2016_09_13__4.8
